app/assets/javascripts/controllers/FileCtrl.coffee

```diff
@@ -17,25 +27,39 @@
 
     # if we haven't allready, load file info from dropbox
     unless file.contents and file.contents.length > 0
-      $scope.loading = true
+      file.loading = true
       FileHandler.get($scope.identity.id, file.path).then((data) ->
         file.contents = data
-        $scope.loading = false
+        # find all the allready added files!
+        _.map data, (remote_file) ->
+          findFile $scope.bruse_files, remote_file
+        file.loading = false
         )
 
   $scope.add = (file) ->
-    # make sure we are not adding a folder
-    unless file.is_dir
-      file.loading = true
-      FileHandler.put($scope.identity.id, file).then((data) ->
-        file.bruse_file = data
-        file.loading = false
-        )
+    file.loading = true
+    FileHandler.put($scope.identity.id, file).then((data) ->
+      # add file to list of existing files
+      $scope.bruse_files.push(data)
+      # append bruse_file to this remote file
+      findFile $scope.bruse_files, file
+      file.loading = false
+      )
 
   $scope.remove = (file) ->
     # send delete request
+    file.loading = true
     if file.bruse_file
       FileHandler.delete($scope.identity.id, file).then((data) ->
         file.bruse_file = data
+        file.loading = false
         )
+
+  findFile = (list_of_files, remote_file) ->
+    # find the first corresponding brusefile. note this -------------------vvv
```

Comment:
Find the first corresponding brusefile.
Does this work through finding a match between remote file and a file in "list of files", this can generate several matches and we only want the first match?